src/Mod/TechDraw/Gui/QGSPage.cpp

```diff
@@ -1224,5 +1228,59 @@ QColor QGSPage::getBackgroundColor()
     return fcColor.asValue<QColor>();
 }
 
+void QGSPage::handleRubberBandSelection()
+{
+    if(!isVKeyPressed && !isFKeyPressed && !isEKeyPressed) {
+        return;  // No filtering keys active
+    }
+
+    QList<QGraphicsItem*> items = selectedItems();
+    QList<QGraphicsItem*>::iterator i;
+    for (i = items.begin(); i != items.end(); ++i) {
+        if(isVKeyPressed && dynamic_cast<QGIVertex*>(*i)) {
+            continue;
+        }
+        else if(isFKeyPressed && dynamic_cast<QGIFace*>(*i)) {
+            continue;
+        }
+        else if(isEKeyPressed && dynamic_cast<QGIEdge*>(*i)) {
+            continue;
+        }
+        (*i)->setSelected(false);
+    }
+}
```

Comment:
Note, that this filtering causes lag even at with selections of just 20 edges... any ideas?

Reply:
One possibility is that QGIVertex, *Face, *Edge are scattered throughout memory hence the lag. Although I'm not sure this would cause the lag of just 20 edges. Perhaps try printing the number of items in QList; maybe it selects much more than 20 entities.

Reply:
@pure-machinery Hmmm, what if QGIVertex is 6 inheritances deep from QGraphicsItem?

Reply:
@benj5378 have you tried disabling QMeasure by changing https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Measure/Gui/QuickMeasure.cpp#L125 to:

`        if (vpType == "SketcherGui::ViewProviderSketch" && vp->isEditing() || vpType == "TechDrawGui::ViewProviderProjGroupItem") {`

it made a world of difference to Sketcher Edit Mode and it not as though it's needed in this workbench.

Reply:
@Syres916 I don't see what you mean by disabling QMeasure?

But I do see, that having a string defining the type at a high level would avoid dynamic_cast and thus huge speed improvements. Further more, we might make it so that filtering is only done if changes to selection - now it's just on mouse movement,